apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1.IntegrationTests/DatastoreDbTest.cs

```diff
@@ -441,13 +442,58 @@ public void Count_WithStructuredQuery()
             var query = new Query("CountTestStQuery");
             AggregationQuery aggQuery = new AggregationQuery(query)
             {
-                Aggregations = { Aggregations.Count("count")}
+                Aggregations = { Count("count") }
             };
             AggregationQueryResults results = db.RunAggregationQuery(aggQuery);
             long count = results["count"].IntegerValue;
             Assert.Equal(2, count);
         }
 
+        [Fact]
+        public void Aggregation_WithGqlQuery()
+        {
+            var db = _fixture.CreateDatastoreDb();
+            var keyFactory = db.CreateKeyFactory("Students");
+            var entities = new[]
+            {
+                new Entity { Key = keyFactory.CreateKey("11"), ["age"] = 12, ["height"] = 5  },
+                new Entity { Key = keyFactory.CreateKey("21"), ["age"] = 12, ["height"] = 4.6  },
+                new Entity { Key = keyFactory.CreateKey("31"), ["age"] = 14, ["height"] = 4  },
+                new Entity { Key = keyFactory.CreateKey("41"), ["age"] = 11, ["height"] = 5.2  }
+            };
+            db.Insert(entities);
+
+            var gql = new GqlQuery { QueryString = "SELECT sum(height), avg(height) as `avgheight` FROM Students" };
+            AggregationQueryResults results = db.RunAggregationQuery(gql);
+            Assert.Equal(18.8, results["property_1"].DoubleValue);
+            Assert.Equal(4.7, results["avgheight"].DoubleValue);
+        }
+
+        [Fact]
+        public void Aggregations_StructuredQuery()
+        {
+            var db = _fixture.CreateDatastoreDb();
+            var keyFactory = db.CreateKeyFactory("Student");
+            var entities = new[]
+            {
+                new Entity { Key = keyFactory.CreateKey("6"), ["age"] = 12, ["height"] = 5  },
+                new Entity { Key = keyFactory.CreateKey("7"), ["age"] = 12, ["height"] = 4.6  },
+                new Entity { Key = keyFactory.CreateKey("8"), ["age"] = 14, ["height"] = 4  },
+                new Entity { Key = keyFactory.CreateKey("9"), ["age"] = 11, ["height"] = 5.2  }
+            };
+            db.Insert(entities);
+
+            var query = new Query("Student");
+            AggregationQuery aggQuery = new AggregationQuery(query)
+            {
+                Aggregations = { Sum("age", "Sum_age"), Average("age"), Count("Count") }
```

Comment:
Super-nit: everywhere else, our aliases are lower-case. I'd suggest `sum_age` and `count`, assuming that's the convention that's normally see in Datastore docs etc. (Likewise, we've got `avgheight` above - is using an underscore conventional or not?)

Not massively important of course, as this is only a test rather than a sample, but it's good to avoid distracting inconsistencies even in tests.

(These two tests are definitely better than the previous large set though - thanks.)

Reply:
+1 to being consistent with alias naming to avoid unnecesary differences. And in particular `avgage` is pretty hard to read.

And as for the tests, I would have the ones in the classes `DatastoreDbTest` and `DatastoreTransactionTest` have the same structure:

- One GQL test that includes the 3 aggregations
- One structured test that includes the 3 aggregations

This is because:

- Clarity. Unnecesary differences are distracting and distractions hinder maintenability. When we land here in a year, we'll be asking ourselves why the tests in `DatastoreDbTest` and `DatastoreTransactionTest` are structured differently when they could be structured the same.
- Testing practices: The 2 types of queries (GQL and structured) are implemented differently, so we should test for them separately. The 3 types of aggregations are implemented the same (for the same type of query) so it's OK that we test them together, at least in integration tests (unit tests should definetely test them separately as well).